runner.py

```diff
@@ -57,49 +63,76 @@
     is_flag=True,
     help="Skips moving output data to reporting folder",
 )
+def cli_qd_start(
+    fastqdir,
+    outdir,
+    sample_name,
+    ss_path,
+    strandedness,
+    testrun,
+    skip_rnaseq,
+    skip_rnafusion,
+    save_reference,
+    skip_report,
+):
+    qd_start(
+        fastqdir=fastqdir,
+        outdir=outdir,
+        sample_name=sample_name,
+        ss_path=ss_path,
+        strandedness=strandedness,
+        testrun=testrun,
+        skip_rnaseq=skip_rnaseq,
+        skip_rnafusion=skip_rnafusion,
+        save_reference=save_reference,
+        skip_report=skip_report)
+
 def qd_start(
-    fastqdir: str,
-    outdir: str,
-    sample_name: str,
-    strandedness: str,
-    testrun: bool,
-    skip_rnaseq: bool,
-    skip_rnafusion: bool,
-    save_reference: bool,
-    skip_report: bool,
+    fastqdir: str = None,
+    outdir: str = None,
+    sample_name: str = None,
+    ss_path: str = None,
+    strandedness: str = None,
+    testrun: bool = False,
+    skip_rnaseq: bool = False,
+    skip_rnafusion: bool = False,
+    save_reference: bool = False,
+    skip_report: bool = False,
+    logger = None,
 ) -> None:
 
     # Read in the config
     config = get_config()
 
     # Set up the logger function
-    now = datetime.datetime.now()
-    logdir = config.get("general", "wrapper_log_dir")
-    os.makedirs(logdir, exist_ok=True)
-    logfile = os.path.join(
-        logdir,
-        "QD-rnaseq-wrapper_" + now.strftime("%y%m%d_%H%M%S") + ".log",
-    )
-    logger = setup_logger("qd-rnaseq", logfile)
-    logger.info("Starting the RNAseq pipepline wrapper.")
+    if not logger:
+        now = datetime.datetime.now()
+        logdir = config.get("general", "wrapper_log_dir")
+        os.makedirs(logdir, exist_ok=True)
+        logfile = os.path.join(
+            logdir,
+            "QD-rnaseq-wrapper_" + now.strftime("%y%m%d_%H%M%S") + ".log",
+        )
+        logger = setup_logger("qd-rnaseq", logfile)
 
     # If testrun, skip fastq handling
     if not testrun:
-        # Make sure fastqdir looks good
-        try:
-            sanitize_fastqdir(fastqdir)
-        except Exception as e:
-            logger.error(e)
+        #Sanity check
+        if fastqdir == None and ss_path == None:
+            logger.error("Please provide either a fastqdir or a samplesheet")
             sys.exit(1)
 
         # Make samplesheet from fastq dir
-        logger.info(f"Creating samplesheet.csv from {fastqdir}")
-        scriptpath = config.get("general", "fastq_to_ss_path")
-        try:
-            ss_path = dir_to_samplesheet(scriptpath, fastqdir, strandedness)
-        except Exception as e:
-            logger.error(e)
-            sys.exit(1)
+        if not ss_path:
+            logger.info(f"No samplesheet provided. Creating samplesheet.csv from {fastqdir}")
```

Comment:
This is the term used in nf-core, that is why this is named so here. But yeah, it does make it a bit confusing since we deal with the Illumina samplesheet stuff. But it is a sheet of samples. :)